CONTRIBUTING.md

```diff
@@ -1,238 +1,191 @@
+  issue being resolved.
 
-## Benchmarking
+In general almost all commits should fall into one of the above categories. In the cases where they don't please submit
+your PR untagged.
 
-Ember Data is instrumented with [heimdalljs](https://github.com/heimdalljs/heimdalljs-lib)
- Top level scenarios for benchmarking are available via the `query` route in
- the dummy app, and desired scenarios to be run can be configured via `benchmarks/config.js`.
+#### Developing a New Feature with Feature Flags
 
- The scenarios are configured to interop with [heimdall-query](https://github.com/heimdalljs/heimdall-query)
- for analysis. To run scenarios:
+Sometimes a new feature will require use of a feature flag.
 
-  1. Start the dummy app with instrumentation on: `ember s --instrument`
+Feature flags allow new features to be tested in dev builds, but
+the features are stripped out of production builds automatically.
 
-  2. Configure `benchmarks/config.js` with desired scenarios
+1. Add your new feature flag to the [config/features.json](https://github.com/emberjs/data/blob/master/config/features.json) file.
 
-  3. To run both the benchmarks and the analysis: `node ./benchmarks`
+```js
+{
+  "ds-boolean-transform-allow-null": null,
+  "ds-mynew-feature": null
+}
+```
 
-      a.) To just collect data (no analysis): `node ./benchmarks/bash-run.js`
-      b.) To just run analysis (w/cached data): `node ./benchmarks/bash-analyze.js`
-      c.) To cache a data set or use a cached data set, all commands accept `-c ./path/to/cache/dir`
+Give it a default of `null` so it will not be used in production builds.
 
-  4. Do not commit cached data results, these should be git ignored already.
+2. Import `isEnabled` from `ember-data/-private`, wrapping any new
+   code with your feature:
 
-# Pull Requests
+```js
+import { isEnabled } from 'ember-data/-private';
 
-We love pull requests. Here's a quick guide:
+if (isEnabled('ds-mynew-feature')) {
+  // ... any additional code
+} else {
+  // ... any previous code that may have been overwritten
+}
+```
 
-1. Fork the repo.
+3. Similarly, you will want to wrap any new or edited tests with the same
+   feature flag.
 
-2. Run the tests. We only take pull requests with passing tests, and it's great
-to know that you have a clean slate, see notes on how to run unit tests [here](https://github.com/emberjs/data#how-to-run-unit-tests). (To see tests in the browser,
-run `npm start` and open `http://localhost:4200/tests`.)
+```js
+import { isEnabled } from 'ember-data/-private';
 
-3. Add a test for your change. Only refactoring and documentation changes
-require no new tests. If you are adding functionality or fixing a bug, we need
-a test!
+if (isEnabled('ds-mynew-feature')) {
+  test('test for new feature', function(assert) {
+    // ...
+  });
+}
+```
 
-4. Make the test pass.
+This will allow the test suite to run as normal.
```

Comment:
I might be a little more descriptive here. Maybe something along the lines of "This will ensure the new tests are only run when then feature is included in Ember Data."